### ISSUE TITLE
Dolt can't handle two single quotes in a default value on create table

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#5411]
Repro:
```
$ dolt sql -q "CREATE TABLE \`test\` ( \`id\` int NOT NULL AUTO_INCREMENT,  \`text\` varchar(255) DEFAULT 'ABC''s test', PRIMARY KEY (\`id\`) ) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb3 ROW_FORMAT=COMPACT;"
$ dolt sql -q "show create table test";
Error parsing SQL
syntax error at position 19 near 'test'
SELECT 'ABC's test'
                   ^
syntax error at position 19 near 'test'
$ dolt sql -q "insert into test values ()" 
Error parsing SQL
syntax error at position 19 near 'test'
SELECT 'ABC's test'
                   ^
syntax error at position 19 near 'test'
```

Workaround is to use double quotes like so:
```
$ dolt sql -q "CREATE TABLE \`test\` ( \`id\` int NOT NULL AUTO_INCREMENT,  \`text\` varchar(255) DEFAULT \"ABC''s test\", PRIMARY KEY (\`id\`) ) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb3 ROW_FORMAT=COMPACT;"
```

**Comment by nitheesh-me** [comment]
@timsehn Thanks for raising the issue on my behalf.

Also, something to note during the `dump`, for the first sample
```console
$ dolt sql -q "CREATE TABLE \`test\` ( \`id\` int NOT NULL AUTO_INCREMENT,  \`text\` varchar(255) DEFAULT 'ABC''s test', PRIMARY KEY (\`id\`) ) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb3 ROW_FORMAT=COMPACT;"
$ dolt dump -r sql --file-name test.sql
Error with dumping test.
cause: syntax error at position 19 near 'test'
```
---
Also, for the workaround, backup and restore are not compatible

```console
$ dolt sql -q "CREATE TABLE \`test\` ( \`id\` int NOT NULL AUTO_INCREMENT,  \`text\` varchar(255) DEFAULT \"ABC''s test\", PRIMARY KEY (\`id\`) ) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb3 ROW_FORMAT=COMPACT;"
$ dolt dump -r sql --file-name test.sql
Successfully exported data.
$ cat test.sql
CREATE DATABASE IF NOT EXISTS `test`; USE `test`;
SET FOREIGN_KEY_CHECKS=0;
SET UNIQUE_CHECKS=0;
DROP TABLE IF EXISTS `test`;
CREATE TABLE `test` (
  `id` int NOT NULL AUTO_INCREMENT,
  `text` varchar(255) DEFAULT 'ABC's test',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin;
$ dolt sql < test.sql
Database changed
Error parsing SQL
syntax error at position 96 near 's'

CREATE TABLE `test` (
  `id` int NOT NULL AUTO_INCREMENT,
  `text` varchar(255) DEFAULT 'ABC's test',
                                     ^
syntax error at position 96 near 's'
```

So, taking a dump and loading it back again will make us lose all data 😟 , and leave us with an empty table, with an error during execution.

**Comment by timsehn** [comment]
This is fixed in 0.75.0
